integration-templates/sharepoint-online/syncs/file-sync.ts

```diff
@@ -0,0 +1,125 @@
+import type { NangoSync, FileMetadata } from '../../models';
+import type { PaginationParams, PaginationResponse } from '../helper/pagination';
+import paginate from '../helper/pagination.js';
+import type { SharePointMetadata } from '../types';
+
+export default async function fetchData(nango: NangoSync): Promise<void> {
+    const metadata = (await nango.getMetadata()) as SharePointMetadata;
+
+    if (!metadata || !metadata.sitesToSync || metadata.sitesToSync.length === 0) {
+        throw new Error(`Metadata empty for connection id: ${nango.connectionId}`);
+    }
+
+    metadata.syncedSites = metadata.syncedSites || [];
+
+    const siteIdToLists = await getSiteIdToLists(nango, metadata.sitesToSync);
+
+    for (const [siteId, listIds] of Object.entries(siteIdToLists)) {
+        for (const listId of listIds) {
+            try {
+                const deltaToken = getDeltaToken(metadata, siteId, listId);
+
+                const params: PaginationParams = {
+                    endpoint: `/v1.0/sites/${siteId}/lists/${listId}/items/delta`,
+                    deltaToken: deltaToken,
+                    $top: 100
+                };
+
+                const newDeltaToken = await processListItems(nango, params, siteId, listId);
+
+                if (newDeltaToken) {
+                    await updateDeltaToken(nango, metadata, siteId, listId, newDeltaToken);
+                }
+            } catch (error) {
+                throw new Error(`Error in fetchData: ${error}`);
```

Comment:
${error} does not stringify correctly it will just output `{}`, it's nice to have a message but I think it's better to let it flow until we have a more robust system in place